weblate/machinery/systran.py

```diff
@@ -0,0 +1,62 @@
+# Copyright © Michal Čihař <michal@weblate.org>
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from .base import MachineTranslation, MachineTranslationError
+from .forms import KeyMachineryForm
+
+
+class SystranTranslation(MachineTranslation):
+    """Systran machine translation support."""
+
+    name = "Systran"
+    max_score = 90
+    settings_form = KeyMachineryForm
+
+    def check_failure(self, response):
+        if "error" not in response:
+            return
+        raise MachineTranslationError(response["error"]["message"])
+
+    def download_languages(self):
+        """Download list of supported languages from a service."""
+        response = self.request(
+            "get",
+            "https://api-translate.systran.net/translation/supportedLanguages",
+            params={"key": self.settings["key"]},
+        )
+        payload = response.json()
+        self.check_failure(payload)
+        return [pair["target"] for pair in payload["languagePairs"]]
```

Comment:
You only extract `target` from the supported pairs. That's why it doesn't recognize pair as supported in the test. You should fetch full pairs and adjust `is_supported` method to work on it. You can basically copy an implementation from Apertium:

https://github.com/WeblateOrg/weblate/blob/18d07a1fb548cb41a30ee0c9d02e8bfcc874d74a/weblate/machinery/apertium.py#L91-L101